src/v1/views/shared/Header.ts

```diff
@@ -104,74 +74,74 @@ export default class Header extends View {
   }
 
   /* eslint complexity: 0 */
-  setBeacon(NextBeacon, options) {
+  private setBeacon(NextBeacon, options): void {
     const selector = '[data-type="beacon-container"]';
     const container = this.$(selector);
-    const transition = typeOfTransition(this.currentBeacon, NextBeacon, options);
+    const transition = this.typeOfTransition(NextBeacon, options);
 
     switch (transition) {
-    case 'none':
-      this.$el.addClass(NO_BEACON_CLS);
-      return;
-    case 'same':
-      return;
-    case 'add':
-      this.$el.removeClass(NO_BEACON_CLS);
-      addBeacon(this, NextBeacon, selector, options);
-      return Animations.explode(container);
-    case 'remove':
-      this.$el.addClass(NO_BEACON_CLS);
-      return Animations.implode(container)
-        .then(() => {
-          removeBeacon(this);
-        })
-        .done(); // TODO: can this be removed if Animations.implode returns standard ES6 Promise?
-    case 'fade':
-      // Other transitions are performed on the beacon container,
-      // but this transition is on the content inside the beacon.
-      // For a SecurityBeacon the username change will update the
-      // AppState and trigger an transition to a new Becon
-      // Since there is no url change this method is not called.
-      // For a FactorBeacon a page refresh has occurred
-      // so we execute the beacon's own transition method.
-      if (!this.currentBeacon.fadeOut) {
-        throw new Error('The current beacon is missing the "fadeOut" method');
-      }
-      options.animate = true;
-      return this.currentBeacon
-        .fadeOut()
-        .then(() => {
-          removeBeacon(this);
-          addBeacon(this, NextBeacon, selector, options);
-        })
-        .done(); // TODO: can this be removed if fadeOut returns standard ES6 Promise?
-    case 'swap':
-      return Animations.swapBeacons({
-        $el: container,
-        swap: () => {
-          const isLoading = isLoadingBeacon(this.currentBeacon);
-
-          // Order of these calls is important for -
-          // loader --> security/factor beacon swap.
-          removeBeacon(this);
-          if (isLoading) {
```

Comment:
is this the fix? That we now remove the classes and we don't need to check if the beacon is loading?

Reply:
Yes, we don't evaluate the state 2 times (if we are in `swap` case the beacon `isLoading`) and the changes in `removeLoadingBeacon` to move the actual update to happen in `unload` case.